api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -101,6 +101,20 @@
     String getJVMOptions(List<DAG.OperatorMeta> operatorMetaList);
   }
 
+  /**
+   * The streaming application master web service authentication strategy.
+   * ENABLE - Enable authentication for web service access
+   * DISABLE - Disable authentication for web services
+   * FOLLOW_HADOOP_AUTH - Follow Hadoop authentication, if hadoop authentication is enabled, i.e., if it is set to something
+   *                other than "simple", enable authentication for web services as well
+   * FOLLOW_HADOOP_HTTP_AUTH - Follow Hadoop HTTP authentication, if hadoop authentication is enabled, i.e., if it is
+   *                set to something other than "simple", enable authentication for web services as well
+   */
+  enum StramHTTPAuthentication
+  {
+    ENABLE, DISABLE, FOLLOW_HADOOP_AUTH, FOLLOW_HADOOP_HTTP_AUTH
```

Comment:
I was thinking about this, all options still use the builtin authentication. The follow_ options determine how this auth is enabled or disable based on whether hadoop auth is enabled. I understand your concern, what would be a good substitute?